app/src/userpages/components/KeyField/index.jsx

```diff
@@ -0,0 +1,113 @@
+// @flow
+
+import React from 'react'
+import copy from 'copy-to-clipboard'
+import cx from 'classnames'
+import { I18n, Translate } from 'react-redux-i18n'
+import TextInput from '$shared/components/TextInput'
+import Meatball from '$shared/components/Meatball'
+import DropdownActions from '$shared/components/DropdownActions'
+
+import KeyFieldEditor from './KeyFieldEditor'
+
+import styles from './keyField.pcss'
+
+type Props = {
+    keyName: string,
+    value?: string,
+    hideValue?: boolean,
+    className?: string,
+    onSave?: (?string, ?string) => void,
+    onDelete?: () => void,
+}
+
+type State = {
+    hidden: boolean,
+    editing: boolean,
+}
+
+class KeyField extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props)
+
+        this.state = {
+            hidden: !!props.hideValue,
+            editing: false,
+        }
+    }
+
+    toggleHidden = () => {
+        this.setState({
+            hidden: !this.state.hidden,
```

Comment:
should probably use the functional form when referencing current state in next state `this.setState(({ hidden }) => ({ hidden: !hidden }))`